### PR TITLE
Fixes #1870, port hash function into Satellite

### LINKED ISSUE #1870: Figure out how to share our hasing algo across microservices
We have a common hashing algorithm we use for a lot of things, defined in src/api/auth/src/hash.js (as well as our current backend), and we need to figure out how to share this.  Probably it needs to get moved to Satellite, so all services can access it.

### PR DESCRIPTION
**Pull request by Metropass**
## Issue This PR Addresses
Fixes #1870

## Type of Change
Ported Hash Functions from Satellite onto Telescope, removed `hash.js`

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
Discussed in #1870 , We ended up porting our hash function onto Satellite, and removed `hash.js` on both `/api` and `/backend`

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,6 +53,7 @@
   "dependencies": {
     "@elastic/elasticsearch": "7.11.0",
     "@elastic/elasticsearch-mock": "0.3.0",
+    "@senecacdot/satellite": "1.8.0",
```
Comment (humphd) [inline package.json:56]: I don't want to use Satellite in the monolithic back-end, only in the microservices.  Let's undo this.

File: src/api/auth/src/admin.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 const User = require('./user');
-const hash = require('./hash');
+const satellite = require('@senecacdot/satellite');
```
Comment (humphd) [inline src/api/auth/src/admin.js:2]: Just pull in the function you need.

```js
const { hash } = require('@senecacdot/satellite');
```

File: src/backend/data/admin.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 const Feed = require('./feed');
 const User = require('./user');
-const hash = require('./hash');
+const satellite = require('@senecacdot/satellite');
```
Comment (humphd) [inline src/backend/data/admin.js:3]: No to this change (only in microservices).

**Review by humphd — CHANGES_REQUESTED** [review]
Bring back src/backend/data/hash.js and this is probably good to go.

**Comment by Metropass** [comment]
> Bring back src/backend/data/hash.js and this is probably good to go.

Just did, and I rebased, so it should be good to go. Are there any other modules that we could port onto Satellite?

**Review by humphd — DISMISSED** [review]

**Review by birtony — DISMISSED** [review]

**Review by humphd — APPROVED** [review]

**Review by birtony — APPROVED** [review]

**Pull request merged**